packages/contracts/contracts/factory/DAOFactory.sol

```diff
@@ -165,135 +162,38 @@ contract DAOFactory {
-    }
-
-    /// @notice Internal helper method to create and setup an `AllowlistVoting` contract for a DAO.
-    /// @param _dao The associated DAO.
-    /// @param _allowlistVoters The array of the allowed voting addresses.
-    /// @param _voteConfig The vote configuration.
-    function createAllowlistVoting(
-        DAO _dao,
-        address[] calldata _allowlistVoters,
-        VoteConfig calldata _voteConfig
-    ) internal returns (AllowlistVoting allowlistVoting) {
-        allowlistVoting = AllowlistVoting(
-            createProxy(
-                allowlistVotingBase,
-                abi.encodeWithSelector(
-                    AllowlistVoting.initialize.selector,
-                    _dao,
-                    _dao.getTrustedForwarder(),
-                    _voteConfig.participationRequiredPct,
-                    _voteConfig.supportRequiredPct,
-                    _voteConfig.minDuration,
-                    _allowlistVoters
-                )
-            )
-        );
-
-        // Grant dao the necessary permissions for AllowlistVoting
-        BulkPermissionsLib.ItemSingleTarget[]
-            memory items = new BulkPermissionsLib.ItemSingleTarget[](4);
-        items[0] = BulkPermissionsLib.ItemSingleTarget(
-            BulkPermissionsLib.Operation.Grant,
-            address(_dao),
-            allowlistVoting.MODIFY_ALLOWLIST_PERMISSION_ID()
-        );
-        items[1] = BulkPermissionsLib.ItemSingleTarget(
-            BulkPermissionsLib.Operation.Grant,
-            address(_dao),
-            allowlistVoting.SET_CONFIGURATION_PERMISSION_ID()
-        );
-        items[2] = BulkPermissionsLib.ItemSingleTarget(
-            BulkPermissionsLib.Operation.Grant,
-            address(_dao),
-            allowlistVoting.UPGRADE_PERMISSION_ID()
-        );
-        items[3] = BulkPermissionsLib.ItemSingleTarget(
-            BulkPermissionsLib.Operation.Grant,
-            address(_dao),
-            allowlistVoting.SET_TRUSTED_FORWARDER_PERMISSION_ID()
-        );
-
-        _dao.bulkOnSingleTarget(address(allowlistVoting), items);
-    }
-
-    /// @notice Internal helper method to set up the required base contracts on `DAOFactory` deployment.
-    function setupBases() private {
-        erc20VotingBase = address(new ERC20Voting());
-        allowlistVotingBase = address(new AllowlistVoting());
-        daoBase = address(new DAO());
+    /// @notice Check if one of the permissions is granting `EXECUTE_PERMISSION_ID`.
+    /// @param permissions List of permissions to be checked.
+    function hasExecutePermission(BulkPermissionsLib.ItemMultiTarget[] memory permissions)
+        internal
+        view
+        returns (bool)
+    {
+        for (uint256 i = 0; i < permissions.length; i++) {
+            if (
+                permissions[i].operation == BulkPermissionsLib.Operation.Grant &&
+                permissions[i].permissionId == DAO(payable(daoBase)).EXECUTE_PERMISSION_ID()
```

Comment:
Problem 1: `DAO(payable(daoBase)).EXECUTE_PERMISSION_ID()` This brings lots of gas for no reason and this line can be done in the constructor !!

Problem 2: This does not effectively detect whether or not plugin is governance plugin or not.

Reply:
will be moved into its own task.